src/middlewared/middlewared/plugins/cluster_linux/ctdb_shared_volume.py

```diff
@@ -0,0 +1,213 @@
+                    'shared volume can be created.'
+                )
+
+            # create the ctdb shared volume
+            req = await self.construct_gluster_ctdb_api_request(con_peers)
+            await(
+                await self.middleware.call('gluster.volume.create', req)
+            ).wait(raise_error=True)
+
+        if not started:
+            # start it if we get here
+            await(
+                await self.middleware.call('gluster.volume.start', self.CTDB_VOL_NAME)
+            ).wait(raise_error=True)
+
+        # try to mount it locally
+        await(
+            await self.middleware.call('ctdb.shared.volume.mount')
+        ).wait(raise_error=True)
+
+        return 'SUCCESS'
+
+    @accepts()
+    @job(lock=CRE_OR_DEL_LOCK)
+    async def delete(self, job):
+        """
+        Delete and unmount the shared volume used by ctdb daemon.
+        """
+
+        # nothing to delete if it doesn't exist
+        exists, started = await self.shared_volume_exists_and_started()
+        if not exists:
+            return
+
+        # umount it first
+        await(
+            await self.middleware.call('ctdb.shared.volume.umount')
+        ).wait(raise_error=True)
+
+        if started:
+            # stop the volume
+            force = {'force': True}
+            await(
+                await self.middleware.call('gluster.volume.stop', self.CTDB_VOL_NAME, force)
+            ).wait(raise_error=True)
+
+        # now delete it
+        data = {'name': self.CTDB_VOL_NAME}
+        await(
+            await self.middleware.call('gluster.volume.delete', data)
+        ).wait(raise_error=True)
+
+        return 'SUCCESS'
+
+    @accepts()
+    @job(lock=JOB_LOCK)
+    async def mount(self, job):
+        """
+        Mount the ctdb shared volume locally.
+        """
+
+        # We call this method on startup so if we were
+        # to add a call to the `create` method then all
+        # nodes in the cluster would try to create the
+        # shared volume which is unnecessary and will
+        # cause lots of errors since only 1 node in the
+        # cluster needs to create the volume and it gets
+        # propagated to the peers that were specified
+        # during the creation of the shared volume.
+        #
+        # In summary....dont try to be fancy and call the
+        # create method in here or it's going to cause
+        # headaches.
+        exists, started = await self.shared_volume_exists_and_started()
+        if not exists or not started:
+            raise CallError(
+                'The ctdb shared volume does not exist or '
+                'is not started.'
+            )
+
+        path = pathlib.Path(CTDBConfig.CTDB_LOCAL_MOUNT.value)
+        try:
+            # make sure the dirs are there
+            path.mkdir(parents=True, exist_ok=True)
+        except Exception as e:
+            raise CallError(f'{e}')
```

Comment:
Just noticed this lonely CallError without any description before the error message